### PR TITLE
[t3118] Prevent certain key presses from firing multiple times on keydow...

### PR DESCRIPTION
**Pull request by mjschranz**
...n

### CODE REVIEW

File: src/ui/ui.js [hunk 1]
```diff
@@ -17,11 +17,15 @@ define( [ "core/eventmanager", "./toggler",
...
-    "INPUT",
-    "VIDEO",
-    "AUDIO"
-  ];
+        "TEXTAREA",
+        "INPUT",
+        "VIDEO",
+        "AUDIO"
+      ],
+      __unwantedHoldKeys = [
```
Comment (ScottDowne) [inline src/ui/ui.js]: Change this variable name to disabledKeyRepeats and add space key to it, maybe?

File: src/ui/ui.js [hunk 2]
```diff
@@ -546,24 +550,35 @@ define( [ "core/eventmanager", "./toggler",
...
+
+    function unbindKeyListeners() {
       window.removeEventListener( "keydown", onKeyDown, false );
+      window.removeEventListener( "keyup", onKeyUp, false );
     }
 
-    function bindKeyDownListener() {
+    function bindKeyListeners() {
       window.addEventListener( "keydown", onKeyDown, false );
+      window.addEventListener( "keyup", onKeyUp, false );
```
Comment (ScottDowne) [inline src/ui/ui.js]: You should only add this event inside the onKeyDown. This way, you have less events being held. Less prone to bugs too.

File: src/ui/ui.js [hunk 3]
```diff
@@ -546,24 +550,35 @@ define( [ "core/eventmanager", "./toggler",
...
     }
 
-    function unbindKeyDownListener() {
+    function onKeyUp() {
+      window.addEventListener( "keydown", onKeyDown, false );
+    }
+
+    function unbindKeyListeners() {
       window.removeEventListener( "keydown", onKeyDown, false );
+      window.removeEventListener( "keyup", onKeyUp, false );
```
Comment (ScottDowne) [inline src/ui/ui.js]: onKeyUp should unbind itself after usage, and not here.

File: src/ui/ui.js [hunk 4]
```diff
@@ -546,24 +550,35 @@ define( [ "core/eventmanager", "./toggler",
       // this allows backspace and del to do the same thing on windows and mac keyboards
       key = key === 46 ? 8 : key;
       if( processKey[ key ] && !eTarget.isContentEditable && __unwantedKeyPressElements.indexOf( eTarget.nodeName ) === -1 ){
+
+        if ( __unwantedHoldKeys.indexOf( key ) > -1 ) {
+          window.removeEventListener( "keydown", onKeyDown, false );
```
Comment (ScottDowne) [inline src/ui/ui.js]: Add the keyUp listener here.

File: src/ui/ui.js [hunk 5]
```diff
@@ -546,24 +550,35 @@ define( [ "core/eventmanager", "./toggler",
...
+        }
+
         processKey[ key ]( e );
       } // if
     }
 
-    function unbindKeyDownListener() {
+    function onKeyUp() {
+      window.addEventListener( "keydown", onKeyDown, false );
```
Comment (ScottDowne) [inline src/ui/ui.js]: Like said under this comment, should remove keyup event here, and not in unbind.

File: src/ui/ui.js (hunk 1 above)
Comment (ScottDowne) [inline src/ui/ui.js]: Actually, consider adding esc and del to this list too.

File: src/core/media.js [hunk 6]
```diff
@@ -110,8 +110,8 @@
       };
 
       this.clear = function(){
-        while( _tracks.length > 0 ){
-          _this.removeTrack( _tracks[ 0 ] );
+        for( var i = _tracks.length - 1; i >= 0; i-- ) {
```
Comment (jbuck) [inline src/core/media.js]: `for (`

Comment (jbuck) [inline src/core/media.js]: Also, can't you just set `_tracks.length = 0;` to clear an array? or does `removeTrack()` do more work than that?

File: src/core/media.js [hunk 7]
```diff
@@ -238,9 +231,10 @@
             "trackeventdeselected"
           ]);
           track.setPopcornWrapper( null );
-          _this.sortTracks();
           track._media = null;
           _this.dispatch( "trackremoved", track );
+          _orderedTracks.splice( orderedIndex, 1 );
```
Comment (jbuck) [inline src/core/media.js]: Wait, we remove a track after we send an event for it? That order seems wrong...

Comment (ScottDowne) [inline src/core/media.js]: Agreed, this seems wrong.

Did you guys discuss this and come to an agreement?

File: src/core/media.js [hunk 8]
```diff
@@ -238,9 +231,10 @@
...
+          _this.sortTracks( orderedIndex );
```
Comment (ScottDowne) [inline src/core/media.js]: Do we need to sort after a splice? Should be fine, no?

Comment (ScottDowne) [inline src/core/media.js]: Ah, I see. It is adding a order to the track event's data. Makes sense.

**Pull request merged**